test/python/tensorflow/tf_unittest_runner.py

```diff
@@ -166,20 +176,24 @@ def regex_walk(dirname, regex_input):
     math_*_*_test
     math*_test
     """
-    if (re.search("\.", regex_input) is None):
+    if (re.search(r'\.', regex_input) is None):
+        # a module name regex was given
         test = regex_input + '.py'
     else:
+        # regex has dot(s) e.g. module.class.testfunc
         test = (re.split("\.", regex_input))[0] + '.py'
     module_list = []
     for path, subdirs, files in os.walk(dirname):
         for name in files:
             if fnmatch(name, test):
-                sys.path.append(path)
+                if path not in sys.path:
+                    sys.path.append(os.path.abspath(path))
                 name = os.path.splitext(name)[0]
                 module_list.append(name)
     if not module_list:
-        print("Test name does not exist")
-        sys.exit(1)
+        print("Test pattern/name does not exist:", regex_input, "dirname",
+              dirname)
+        #sys.exit(1)
```

Comment:
commented code
Shouldn't the script exit f the test pattern is not found?

Reply:
No, it's going to excuse and move on and parse more lines from the manifest. This will happen if user gives a pattern which doesn't match with any test.

Reply:
So if the pattern given by user does not match, the script continues with parsing the manifest - Shouldn't it exit because the user specified a certain pattern which wasn't found?

Reply:
I am leaning with the design that should continue parsing. The test result will display all the tests being executed, so user will know if the desired test is being run.

Reply:
As a user I would expect the script to only execute the tests that match the pattern I provide and if none match then not execute anything unless I do not provide a pattern at all, in which case it executes all the tests. It should continue parsing and executing if a test fails in-between in both cases.

Reply:
Let's discuss this over a call